lib/autoupdate/tools/utils.go

```diff
@@ -209,11 +214,22 @@ func teleportPackageURLs(ctx context.Context, uriTmpl string, baseURL, version s
 		flags |= autoupdate.FlagEnterprise
 	}
 
-	teleportURL, err := autoupdate.MakeURL(uriTmpl, baseURL, autoupdate.DefaultPackage, version, flags)
+	// TODO(vapopov): DELETE in v22.0.0 version check - the separate `teleport-tools` package
+	// will be included in all supported versions.
+	pkg := autoupdate.DefaultPackage
+	if runtime.GOOS == constants.DarwinOS &&
+		(semVersion.Major == 17 && semVersion.Compare(*semver.New("17.7.2")) >= 0 ||
+			semVersion.Major == 18 && semVersion.Compare(*semver.New("18.1.5")) >= 0 ||
+			semVersion.Major > 18) {
```

Comment:
Unless some linter complains about the unkeyed fields (also, eventually the wanted semVersion is always going to be higher than 18 so checking that first is very slightly better):

```suggestion
		(semVersion.Major > 18 ||
			semVersion.Major == 18 && semVersion.Compare(semver.Version{18, 1, 5}) >= 0 ||
			semVersion.Major == 17 && semVersion.Compare(semver.Version{17, 7, 2}) >= 0) {
```